### ISSUE TITLE
Runtime Error on Deleting Snowflake Procedure After Updating to 0.51.0

### ISSUE THREAD

**Issue by shaariqch-carma** [pulumi/pulumi-snowflake#568]
### What happened?

After updating the Pulumi Snowflake provider from version 0.46.1 to 0.51.0, I am encountering a runtime error when attempting to delete a Snowflake procedure. The error message indicates an "index out of range" issue within the NewSchemaObjectIdentifierFromFullyQualifiedName function.

Very strange since this was working perfectly when I was on 0.46.1.

### Example

Here is the output of the logs:

```
panic: runtime error: index out of range [1] with length 1
    goroutine 555 [running]:
    github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk.NewSchemaObjectIdentifierFromFullyQualifiedName({0x14000c1a2c0?, 0x100c62e18?})
        /home/runner/go/pkg/mod/github.com/!snowflake-!labs/terraform-provider-snowflake@v0.88.0/pkg/sdk/identifier_helpers.go:215 +0x344
    github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources.DeleteContextProcedure({0x102f21398, 0x1400013e5b0}, 0x1400093be00, {0x102a16900?, 0x14000c11738?})
        /home/runner/go/pkg/mod/github.com/!snowflake-!labs/terraform-provider-snowflake@v0.88.0/pkg/resources/procedure.go:703 +0xfc
    github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).delete(0x14000e415e0, {0x102f212f0, 0x14000fb0630}, 0xd?, {0x102a16900, 0x14000c11738})
        /home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20240229143312-4f60ee4e2975/helper/schema/resource.go:834 +0xe8
    github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14000e415e0, {0x102f212f0, 0x14000fb0630}, 0x14000fb28f0, 0x1400093ba80, {0x102a16900, 0x14000c11738})
        /home/runner/go/pkg/mod/github.com/pulumi/terraform-plugin-sdk/v2@v2.0.0-20240229143312-4f60ee4e2975/helper/schema/resource.go:883 +0x470
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2.v2Provider.Apply({0x1400009d860?, {0x0?, 0x0?, 0x0?}}, {0x102f212f0, 0x14000fb0630}, {0x1024ebb4e, 0x13}, {0x102f21a68?, 0x1400130a0f0}, ...)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.80.0/pkg/tfshim/sdk-v2/provider.go:122 +0x148
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge.(*Provider).Delete(0x140001f0c80, {0x102f212f0?, 0x14000fb0240?}, 0x1400009ce40)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.80.0/pkg/tfbridge/provider.go:1379 +0x4d8
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Delete_Handler.func1({0x102f212f0, 0x14000fb0240}, {0x102dae9c0?, 0x1400009ce40})
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.112.0/proto/go/provider_grpc.pb.go:647 +0x74
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x102f212f0, 0x14000f6fec0}, {0x102dae9c0, 0x1400009ce40}, 0x1400067b660, 0x14000beb8f0)
        /home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x2e8
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Delete_Handler({0x102e99220?, 0x140001f0c80}, {0x102f212f0, 0x14000f6fec0}, 0x140000e7400, 0x1400004d0c0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.112.0/proto/go/provider_grpc.pb.go:649 +0x12c
    google.golang.org/grpc.(*Server).processUnaryRPC(0x14000940000, {0x102f212f0, 0x14000f6fe30}, {0x102f36f40, 0x140012284e0}, 0x14000fa6360, 0x14000f56810, 0x1042b0688, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1386 +0xba0
    google.golang.org/grpc.(*Server).handleStream(0x14000940000, {0x102f36f40, 0x140012284e0}, 0x14000fa6360)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1797 +0xc70
    google.golang.org/grpc.(*Server).serveStreams.func2.1()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1027 +0x8c
    created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 83
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.62.1/server.go:1038 +0x150

  snowflake:index:Procedure (A_PROCEDURE):
    error: error reading from server: EOF

```

### Output of `pulumi about`

```
Dependencies:
NAME               VERSION
@pulumi/gcp        7.18.0
@pulumi/pulumi     3.113.0
@pulumi/snowflake  0.51.0
remeda             1.59.0
@types/node        18.19.31
prettier           3.2.5
```

The rest is redacted for privacy.

### Additional context

Using a custom GCP backend.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by shaariqch-carma** [comment]
I have also tried to:
- delete the resource from state
- import it using what's in the docs:

```
pulumi import snowflake:index/procedure:Procedure A_PROCEDURE 'DATABASE|SCHEMA|A_PROCEDURE|INT'
```

But I get the same error.

_again, specific names redacted for privacy._

**Comment by VenelinMartinov** [comment]
I've reproed this with the code at the bottom. I've also found a workaround:

Importing the resource in the following way works:
```
pulumi import --stack test1 snowflake:index/procedure:Procedure example "db-fbb462f.schema-97c8b1f.proc1(VARCHAR,DATE)"
```
e.g.

```
pulumi import --stack test1 snowflake:index/procedure:Procedure example "DB_NAME.SCHEMA_NAME.PROC_NAME(ARG1,ARG2)"
```

There's a few things going wrong here:

1. Seems like at some point the TF provider changed the way they identify resources.
2. The import string is incorrectly specified in both our docs and the upstream TF docs: https://www.pulumi.com/registry/packages/snowflake/api-docs/procedure/#import lists it as `'dbName|schemaName|procedureName|varchar-varchar-varchar'`, when in reality the code assumes . separated identifiers.
3. The TF provider handles these errors quite badly, there's quite a few issues upstream about it: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues?q=is%3Aissue+is%3Aopen+index+out+of+range


Repro code:
<details>

`Pulumi.yaml`:

```yaml
name: snowflake_panic_yaml
runtime: yaml
description: A minimal Pulumi YAML program
resources:
  db:
    type: snowflake:Database
  schema:
    type: snowflake:Schema
    properties:
      database: ${db.name}
      dataRetentionDays: 1
  proc:
    type: snowflake:Procedure
    properties:
      name: "proc${procName.result}"
      database: ${db.name}
      schema: ${schema.name}
      language: JAVASCRIPT
      arguments:
        - name: arg1
          type: varchar
        - name: arg2
          type: DATE
      comment: Procedure with 2 arguments
      returnType: VARCHAR
      executeAs: CALLER
      returnBehavior: IMMUTABLE
      nullInputBehavior: RETURNS NULL ON NULL INPUT
      statement: |
        var X=1
        return X
  procName:
    type: random:RandomString
    properties:
      length: 10
      special: false

outputs:
  dbId: ${db.id}
  schemaId: ${schema.id}
  procId: ${proc.id}
  procUrn: ${proc.urn}
```

`repro.sh`:
```bash
#!/bin/bas
set -euxo pipefail

pulumi stack init test1 || True
pulumi up --stack test1 --yes --refresh
eval "$(pulumi stack output --shell)"

pulumi state delete $procUrn --yes --stack test1

pulumi import snowflake:index/procedure:Procedure example "${dbId}|${schemaId}|proc1|VARCHAR-DATE"
```

</details>


I'll raise an issue about patching the docs.

**Comment by VenelinMartinov** [comment]
Looks like https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2450 changed the id semantics, not obviously intentional.

Reproed it in TF:

<details>

```terraform
terraform {
  required_providers {
    snowflake = {
      source  = "Snowflake-Labs/snowflake"
      version = "0.88.0"
    }
  }
}

provider "snowflake" {
}

resource "snowflake_database" "db" {
  name                = "MYDB-2"
}

resource "snowflake_schema" "schema" {
  database            = snowflake_database.db.name
  name                = "MYSCHEMA-2"
  data_retention_days = 1
}

# resource "snowflake_procedure" "proc" {
#   name     = "SAMPLEPROC3"
#   database = snowflake_database.db.name
#   schema   = snowflake_schema.schema.name
#   language = "JAVASCRIPT"
#   arguments {
#     name = "arg1"
#     type = "varchar"
#   }
#   arguments {
#     name = "arg2"
#     type = "DATE"
#   }
#   comment             = "Procedure with 2 arguments"
#   return_type         = "VARCHAR"
#   execute_as          = "CALLER"
#   null_input_behavior = "RETURNS NULL ON NULL INPUT"
#   statement           = <<EOT
# var X=1
# return X
# EOT
# }

import {
  to = snowflake_procedure.example
  id = "MYDB-2|MYSCHEMA-2|SAMPLEPROC3|VARCHAR-DATE"
}
```

</details>

**Comment by VenelinMartinov** [comment]
Confirmed the change was intentional: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2722#issuecomment-2061936246

They've updated the docs and are releasing the fix today. We'll pick this up once we upgrade to that upstream version.

**Comment by VenelinMartinov** [comment]
Just kicked off a release, this will get fixed in the docs with v0.52.0
